lib/medtronic/processData.js

```diff
@@ -638,19 +615,170 @@ var buildBasalRecords = function (records) {
   return postrecords;
 };
 
+function buildPrimeRecords(records) {
+
+  var primeRecords = filterHistory([RECORD_TYPES.PRIME], records);
+  var postrecords = [];
+
+  primeRecords.forEach(function (primedatum) {
+
+    console.log('prime:',primedatum.jsDate, common.bytes2hex(primedatum.head), common.bytes2hex(primedatum.date), common.bytes2hex(primedatum.body));
+
+    var amount = struct.extractByte(primedatum.head,4) / 10.0;
+    var fixed  = struct.extractByte(primedatum.head,2) / 10.0;
+
+    var type = fixed === 0 ? 'tubing' : 'cannula';
+
+    var deviceEvent = cfg.builder.makeDeviceEventPrime()
+      .with_deviceTime(sundial.formatDeviceTime(primedatum.jsDate))
+      .with_primeTarget(type)
+      .with_volume(amount)
+      .set('index', primedatum.index);
+
+    if(type === 'cannula' && fixed !== amount) {
+      // it's possible to cancel a cannula prime,
+      // storing the programmed amount in payload for now
+      deviceEvent.set('payload', {programmedAmount: fixed});
+    }
+
+    cfg.tzoUtil.fillInUTCInfo(deviceEvent, primedatum.jsDate);
+    deviceEvent = deviceEvent.done();
+    postrecords.push(deviceEvent);
+  });
+  return postrecords;
+}
+
+function buildRewindRecords(records) {
+
+  var rewindRecords = filterHistory([RECORD_TYPES.REWIND], records);
+  var postrecords = [];
+
+  rewindRecords.forEach(function (rewinddatum) {
+
+    console.log('rewind:',rewinddatum.jsDate, common.bytes2hex(rewinddatum.head), common.bytes2hex(rewinddatum.date), common.bytes2hex(rewinddatum.body));
+
+    var deviceEvent = cfg.builder.makeDeviceEventReservoirChange()
+      .with_deviceTime(sundial.formatDeviceTime(rewinddatum.jsDate))
+      .set('index', rewinddatum.index);
+    cfg.tzoUtil.fillInUTCInfo(deviceEvent, rewinddatum.jsDate);
+
+    deviceEvent = deviceEvent.done();
+    postrecords.push(deviceEvent);
+  });
+  return postrecords;
+}
+
+function buildTimeChangeRecords(records, mostRecent) {
```

Comment:
K, just be aware that the bootstrapping may be "fun" for the CGM data 😉 . At least for how we got the data through the CSV, we had to bootstrap the CGM stream of data separately because it's separate on the device - different index and records it's own time changes.